### DISCUSSION TITLE
Quotes in resource names

### DISCUSSION THREAD

**Discussion by camaya** [Snowflake-Labs/terraform-provider-snowflake#558] [Q&A] [answered]
Hi,

I have a database that was created without terraform that is called `poc_db`. Now I want to create a schema on that database using the terraform provider, this is how my code looks like:

```terraform
resource "snowflake_schema" "stage" {
  database = "poc_db"
  name     = "stage"

  is_managed   = true
  is_transient = false
}
```

This fails with the following error: `Database '"poc_db"' does not exist or not authorized.`

I know that the error is happening because the provider is using quotes for the database name when trying to create the schema so I've got a couple of questions:

1. Is there any way to disable that behaviour?
2. Why are the names of the resources created with quotes?

Cheers!

**Comment by camaya** [comment]
I was looking at old issues and this behaviour sounds similar to this issue: https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/29

**Comment by camaya** [comment, marked as answer]
As most of the times, the answer to this was on the documentation, exactly here: https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html#controlling-case-using-the-quoted-identifiers-ignore-case-parameter

Basically, by default `"test"` and `test` are treated differently and if you want both of them to be treated as equals, you need to set this parameter in Snowflake: `QUOTED_IDENTIFIERS_IGNORE_CASE = true`. The parameter can be set at different levels, check https://docs.snowflake.com/en/sql-reference/parameters.html#session-parameters.

I use a user specifically for Terraform so I changed the parameter for that user like this:

`ALTER USER my_user SET QUOTED_IDENTIFIERS_IGNORE_CASE = true;`

**Comment by nathan-protempo** [comment]
"I have a database that was created without terraform that is called poc_db" - I think your issue is actually that your database is NOT called "poc_db". Snowflake object names are case-sensitive. If the database was created without the name being quoted then the default behaviour of Snowflake will have been to convert that name to uppercase, so the database may actually be called POC_DB. 

When you reference the database name the default behaviour of Snowflake is to convert all references to upper case, such that poc_db, Poc_Db and POC_DB are all treated as POC_DB. However if the reference is quoted, which is always is when using the Snowflake provider, then it's treated literally, i.e. "poc_db" is interpreted as lower case and will not match if the actual name is "POC_DB".

So my suggestion is to confirm what the actual names of your schema objects are, including case, and then use the correct name in your Terraform script. Alternatively, if you know all schema objects are using upper case names consistently you can wrap some upper() functions around variables, names etc. to ensure they will match even if a lower case value is passed in.
